---
title: "'Catch' 無法攔截類型 '<typename>'，因為它不是 'System.Exception' 或繼承自 'System.Exception' 的類別"
ms.date: 07/20/2015
f1_keywords:
- vbc30392
- bc30392
helpviewer_keywords:
- BC30392
ms.assetid: 1d513d1d-38f5-4b4e-95bb-9f1209553803
ms.openlocfilehash: 778591653948b64b1f6336bad6f48d520b22e51c
ms.sourcegitcommit: f8c270376ed905f6a8896ce0fe25b4f4b38ff498
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 06/04/2020
ms.locfileid: "84397125"
---
# <a name="catch-cannot-catch-type-typename-because-it-is-not-systemexception-or-a-class-that-inherits-from-systemexception"></a>'Catch' 無法攔截類型 '\<typename>'，因為它不是 'System.Exception' 或繼承自 'System.Exception' 的類別
`Catch` 只能攔截例外狀況，而您已嘗試攔截非衍生自例外狀況的項目。  
  
 **錯誤 ID︰** BC30392  
  
## <a name="to-correct-this-error"></a>更正這個錯誤  
  
1. 移除 `Catch` 陳述式，或將 `Catch` 的目標變更為實際例外狀況。  
  
## <a name="see-also"></a>另請參閱

- [Try...Catch...Finally 陳述式](../language-reference/statements/try-catch-finally-statement.md)
